app/models/manageiq/providers/autosde/storage_manager/host_initiator.rb

```diff
@@ -0,0 +1,21 @@
+class ManageIQ::Providers::Autosde::StorageManager::HostInitiator < ::HostInitiator
+  supports :create
+
+  def self.raw_create_host_initiator(_ext_management_system, _options = {})
+    host_initiator_to_create = _ext_management_system.autosde_client.StorageHostCreate(
+      :name           => _options['name'],
+      :port_type      => _options['port_type'],
+      :storage_system => PhysicalStorage.find(_options['physical_storage_id']).ems_ref,
+      :iqn            => _options['iqn'] || "",
+      :wwpn           => _options['wwpn'] || "",
+      :chap_name      => _options['chap_name'] || "",
+      :chap_secret    => _options['chap_secret'] || ""
+    )
+
+    begin
+      _ext_management_system.autosde_client.StorageHostApi.storage_hosts_post(host_initiator_to_create)
+    ensure
+      EmsRefresh.queue_refresh(_ext_management_system)
```

Comment:
You sure you want to refresh even if this fails?  I would have thought the opposite, only refresh if the create succeeded.

Is there a timing issue here?  Is the storage_host guaranteed to be created once this API call returns or does it just initiate a process to create it?

Reply:
@llivne I still don't think you want to refresh the provider if creating the storage_host fails since there would be no change in the inventory.

Reply:
Agree. Fixed.